ethereum/core/src/main/java/org/hyperledger/besu/ethereum/transaction/TransactionSimulator.java

```diff
@@ -291,6 +283,38 @@ public Optional<TransactionSimulatorResult> processWithWorldUpdater(
     return Optional.of(new TransactionSimulatorResult(transaction, result));
   }
 
+  private long calculateSimulationGasCap(
+      final CallParameter callParams, final BlockHeader blockHeaderToProcess) {
+    final long simulationGasCap;
+
+    // when not set gas limit is -1
+    if (callParams.getGasLimit() >= 0) {
+      if (rpcGasCap > 0 && callParams.getGasLimit() > rpcGasCap) {
```

Comment:
I wonder if it is more clear to use  `rpcGasCap > 0` as the first test.
Basically, we start by asking if the user have a configured rpcGasCap, and depending on this user choice, we can check other options.

Reply:
I understand it could be subjective, I prefer that user provided gas limit drives the choice.